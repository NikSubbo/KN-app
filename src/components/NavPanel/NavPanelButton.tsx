import { FC } from 'react';
import cx from 'classnames';

type Props = {
  number: number;
  label: string;
};

export const NavPanelButton: FC<Props> = ({ number, label }) => {
  const isHighlighted =
    number === 1 ? 'text-aqua' : 'text-gray cursor-not-allowed';

  return (
    <button
      className={cx(
        'pt-4 pb-3 px-6 text-sm font-medium text-aqua border-b-2',
        isHighlighted
      )}
    >
      {number} {label}
    </button>
  );
};

export default NavPanelButton;
