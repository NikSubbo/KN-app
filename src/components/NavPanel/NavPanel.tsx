import { FC } from 'react';
import NavPanelButton from './NavPanelButton';

export const NavPanel: FC = () => {
  const btnLabeles = ['Product', 'Addressess', 'Overview'];

  return (
    <div className="mt-9 mb-6">
      {btnLabeles.map((el, i) => {
        return <NavPanelButton key={el} number={i + 1} label={el} />;
      })}
    </div>
  );
};

export default NavPanel;
