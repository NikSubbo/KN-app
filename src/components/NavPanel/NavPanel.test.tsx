import { render } from '@testing-library/react';
import NavPanel from './NavPanel';

describe('NavPanel', () => {
  it('should match the snapshot', () => {
    const { container } = render(<NavPanel />);
    expect(container).toMatchSnapshot();
  });
});
