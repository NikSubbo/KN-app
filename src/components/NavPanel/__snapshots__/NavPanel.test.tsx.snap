// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavPanel should match the snapshot 1`] = `
<div>
  <div
    class="mt-9 mb-6"
  >
    <button
      class="pt-4 pb-3 px-6 text-sm font-medium text-aqua border-b-2 text-aqua"
    >
      1
       
      Product
    </button>
    <button
      class="pt-4 pb-3 px-6 text-sm font-medium text-aqua border-b-2 text-gray cursor-not-allowed"
    >
      2
       
      Addressess
    </button>
    <button
      class="pt-4 pb-3 px-6 text-sm font-medium text-aqua border-b-2 text-gray cursor-not-allowed"
    >
      3
       
      Overview
    </button>
  </div>
</div>
`;
